seed/java-sdk/pagination/src/main/java/com/seed/pagination/core/pagination/SyncPagingIterable.java

```diff
@@ -0,0 +1,52 @@
+package com.hume.api.core.pagination;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class SyncPagingIterable<T> extends SyncPage<T> implements Iterable<T> {
+
+    public SyncPagingIterable(boolean hasNext, List<T> items, Supplier<? extends SyncPage<T>> getNext) {
+        super(hasNext, items, getNext);
+    }
+
+    public SyncPagingIterable(boolean hasNext, Optional<List<T>> items, Supplier<? extends SyncPage<T>> getNext) {
```

Comment:
made this extra constructor to handle an `Optional` `items` for 2 reasons:
1. imo feels most important to keep client methods clean over `core` classes, and this basically delegates some extra logic to this class instead of the client method. i'd imagine people spend more time looking at those
2. a bit easier to implement this way!